src/main.cpp

```diff
@@ -1,6 +1,51 @@
-#include "testh.hpp"
+#include <cstdint>
+#include <iostream>
 
+#include "networking/Tcp.hpp"
+
+// Simple recv test for TCP interface
 int main() {
-    foo();
+    // Try to open a socket for listening
+    Tcp::ListenSocket liSock;
+    try {
+        liSock = Tcp::ListenSocket(1234);
+        liSock.listen();
+    } catch (Tcp::OpFailureException&) {
+        std::cerr << "Could not create/open listening socket" << std::endl;
+        return -1;
+    }
+
+    Tcp::ConnSocket coSock;
+    while (true) {
+        // Try to accept an incoming request
+        try {
+            coSock = liSock.accept();
+        } catch (Tcp::OpFailureException&) {
+            std::cerr << "Unable to accept a connection" << std::endl;
+            return -1;
+        }
+        std::cout << "Connected to client!\n"
```

Comment:
Also for std::couts, we can have a config variable that can toggle console output for debug. i.e DEBUG=True

Reply:
Like above, I would just make this a separate module. But cleaner logging would certainly be a plus. We could do something like the original mk1-1-engine-controller, with logger.info(), logger.debug(), logger.error(), etc., and with a #define DEBUG or similar to turn each level on or off.